File: app/src/main/java/com/binaracademy/myaccountant/ui/income/IncomeActivity.java
package com.binaracademy.myaccountant.ui.income;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.binaracademy.myaccountant.R;

public class IncomeActivity extends AppCompatActivity {
    EditText input_gaji;
    Button button_save;
    SharedPreferences sharedPreferences;

    private static final String SHARED_PREF_NAME = "income";
    private static final String KEY_INCOME = "gaji";

    @Override
    protected void onCreate( Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_income);

        input_gaji = findViewById(R.id.input_income);
        button_save = findViewById(R.id.btn_save);
        sharedPreferences = getSharedPreferences(SHARED_PREF_NAME, MODE_PRIVATE);

        button_save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                SharedPreferences.Editor editor =  sharedPreferences.edit();
                editor.putString(KEY_INCOME, input_gaji.getText().toString());
                editor.apply();

//                Intent intent = new Intent(IncomeFragment.this.ProfileFragment.class);
//                startActivity(intent);

                Toast.makeText(IncomeActivity.this, "Gaji Sudah Diinputkan!", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/binaracademy/myaccountant/ui/income/IncomeActivity.kt
package com.binaracademy.myaccountant.ui.income

import android.app.Activity
import android.content.SharedPreferences
import android.os.Bundle
import android.view.inputmethod.InputMethodManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.binaracademy.myaccountant.data.enums.UserType
import com.binaracademy.myaccountant.data.presenter.InitialContract
import com.binaracademy.myaccountant.data.presenter.InitialPresenter
import com.binaracademy.myaccountant.databinding.ActivityIncomeBinding
import com.binaracademy.myaccountant.ui.main.MainActivity
import com.binaracademy.myaccountant.util.helpers.Global
import com.binaracademy.myaccountant.util.helpers.SharedPreferencesManager
import com.binaracademy.myaccountant.util.helpers.intentTo
import kotlinx.coroutines.launch

class IncomeActivity : AppCompatActivity(), InitialContract.View {
	
	private val presenter = InitialPresenter()
	
	private val binding: ActivityIncomeBinding by lazy {
		ActivityIncomeBinding.inflate(layoutInflater)
	}
	
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(binding.root)
		presenter.setView(this)
		
		val userTable = Global.USER_TABLE
		val type = Global.TYPE
		
		val sharedPreferences = SharedPreferencesManager(this, userTable)
		
		val userType = when (sharedPreferences.getString(type, "")) {
			"Si Life Balance" -> UserType.LIFE_BALANCE
			"Si Paling Hemat" -> UserType.PALING_HEMAT
			"Si Tukang Shopping" -> UserType.TUKANG_SHOPPING
			else -> UserType.LIFE_BALANCE
		}
		
		binding.btnSave.setOnClickListener {
			val income = binding.inputIncome.text
			
			lifecycleScope.launch {
				presenter.saveInitialIncome(userType, income.toString().toLong())
			}
		}
		
		binding.root.setOnClickListener {
			hideSoftKeyboard(this)
		}
	}
	
	private fun hideSoftKeyboard(activity: Activity) {
		val inputMethodManager = activity.getSystemService(
			INPUT_METHOD_SERVICE
		) as InputMethodManager
		if (inputMethodManager.isAcceptingText) {
			inputMethodManager.hideSoftInputFromWindow(
				activity.currentFocus!!.windowToken,
				0
			)
		}
	}
	
	override fun onSaveInitialError(error: String) {
		Toast.makeText(this, error, Toast.LENGTH_SHORT).show()
	}
	
	override fun onSaveInitialSuccess(amount: Long) {
		intentTo(MainActivity::class.java)
	}
}